src/include/daos_fs.h

```diff
@@ -1171,6 +1171,20 @@ enum {
 int
 dfs_cont_check(daos_handle_t poh, const char *cont, uint64_t flags, const char *name);
 
+/**
+ * Scan the DFS namespace and report statistics about file/dir size and namespace structure.
+ *
+ * \param[in]	poh	Open pool handle.
+ * \param[in]	cont	POSIX container label.
+ * \param[in]	flags	Unused flag, added for future use.
+ * \param[in]	name	Optional subdirectory path to scan from.
+ *			Start from container root if not specified.
+ *
+ * \return		0 on success, errno code on failure.
+ */
+int
+dfs_cont_scan(daos_handle_t poh, const char *cont, uint64_t flags, const char *name);
```

Comment:
A public C function which is called from C but reports it's output to stdout it's much use IMO, this would be better passing out a scan_args strut.  I think in general this functionality is a great addition but perhaps we should start with the go interface and not commit to a C api until we can have a supportable API.